main.py

```diff
@@ -195,30 +198,79 @@ def main():
                 sg.popup(
                     f"Amiibo encryption key(s) are missing.\nPlease select keys using Settings > Select Key",
                     title="Missing Key!")
+
+        elif event == 'Open JSON':
+            base_dump = bytearray([0] * 540)
+
+            path = filedialog.askopenfilename(filetypes=(('JSON files', '*.json'),))
+            with open(path, encoding="utf-8") as ryujinx_json:
+                to_dump = json.load(ryujinx_json)
+
+            base_dump[84:92] = bytes.fromhex(to_dump['AmiiboId'])
+            base_dump[0x130:0x208] = base64.b64decode(to_dump['ApplicationAreas'][0]['ApplicationArea'])
+            base_dump[0x10a:0x10e] = to_dump['ApplicationAreas'][0]['ApplicationAreaId'].to_bytes(4, 'big')
+
+            if 'Name' in to_dump:
+                utf16 = to_dump["Name"].encode('utf-16-be')
+                base_dump[0x020:0x034] = utf16.ljust(20, b'\x00')
+            else:
+                utf16 = "AMIIBO".encode('utf-16-be')
+                base_dump[0x020:0x034] = utf16.ljust(20, b'\x00')
+            with open('temp.bin', "wb") as temp:
+                temp.write(base_dump)
+            amiibo = VirtualAmiiboFile("temp.bin", config.read_keys(), False)
+            amiibo.randomize_sn()
+            ryujinx_loaded = True
```

Comment:
Instead of doing it this way, you could check to see whether the user picked a bin or a json file in then normal open. If they picked a bin, proceed normally. If they picked a json file, amiibo = JSONVirtualAmiiboFile() and proceed normally. All of this code would get moved to open_bin in the child class